Exercises no longer ask developers to 'write a function' (#2396) (#2406)
exercism/problem-specifications: exercises/nucleotide-codons/description.md

# Description

> Return the name of an amino acid encoded by a particular codon, possibly using shorthand.

Reviewer: "Return" is function specific terminology.

> Translate a given codon into the name of the amino acid that it encodes, possibly using shorthand.

Reviewer: If we're rewording this, the exercise asks you to translate a sequence of codons into a sequence of amino acids. There's no shorthands in any of the tracks that I've seen.

Author: I think that's explained in the description. DNA has the 4 bases, Adenine, Cytine, Thymine, Guanine. The shorthand it talks about is to do with the fact that multiple codons can map to the same amino acid, e.g. AAA and AAG both map to Lysine. So the shorthand they came up with is AAR (R means A|G). I think the "may use shorthand" language is due to the fact that the input might be "AAA", "AAG", or "AAR" and the expected output would be "Lysine"? But I haven't done the exercise yet.

Author: Sorry, wrote that from phone while horizontal and in pain 😅. Had a quick chance to search, and -- because the exercise is deprecated? -- I've only found it (locked) on the Rust track. Because of that and the fact that there's no canonical-data.json file I can't confirm my assumption above is correct. 

Since the exercise is deprecated, is it worth further effort to refine the text?

Reviewer 2: > Since the exercise is deprecated, is it worth further effort to refine the text?

Not really

> Write some code which will output the name of an amino acid encoded by a particular codon, possibly using shorthand.

Reviewer 3: Perhaps

```suggestion
Produce the name of an amino acid encoded by a particular codon, possibly using shorthand.
```

or, more radically,

```suggestion
Translate a given codon into the name of the amino acid that it encodes, possibly using shorthand.
```

Author: I'm going to Bravely Fix it to be like your second suggestion. That reworking had occurred to me too but I was being cautious and trying to use as light a touch as possible. 😅
